text/0000-postgres-major-upgrade-1.md

```diff
@@ -0,0 +1,56 @@
+- Feature Name: `PostgreSQL major version upgrade - preparations`
+- Start Date: 2023-09-30
+- RFC PR: (leave this empty)
+- Bundler Issue: (leave this empty)
+
+# Summary
+
+Phase 1 of the PostgreSQL upgrade process for RubyGems.org revolves around making the necessary preparations to transition from version 11 to version 12 with minimal disruptions.
+
+# Motivation
+
+The key goal of this phase is to ensure that all crucial tools, configurations, and scripts are in place, facilitating a seamless migration from PostgreSQL 11 to 12. By making the most of the mandatory downtime for enabling replication, we also include the "pg_stat_statements" for enhanced monitoring, maximizing the value derived from this period.
+
+# Guide-level explanation
+
+The initial phase of the PostgreSQL upgrade will include:
+
+1. Setting up `pgbouncer` in the Kubernetes cluster with associated services to manage connections.
```

Comment:
will it be set up in the `rubygems.org` repo, and then in the per-environment namespaces? or globally in terraform?

Reply:
IMHO it is enough to run a "maintenance" pod within cluster with `pgbouncer` installed and use it for migration. It will be needed only for few minutes for migration switch as a proxy holding connections opened.